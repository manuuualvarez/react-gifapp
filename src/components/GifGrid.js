import React from 'react'
// Components
import {  GifGridItem } from './GifGridItem'
// Hooks
import { useFetchGifs } from '../hooks/useFetchGifs'

export const GifGrid = ( {category }) => {

    const { data:images, loading } = useFetchGifs(category);

    return (
        <>
            <h3> { category } </h3>
        
            { loading && <p className="animate__animated animate__flash">Loading...</p>  }

            <div className="card-grid">
                {  images.map( img =>  

                    <GifGridItem  
                        key={ img.id }
                        { ...img }
                    />       
                    )
                }
            </div>
        </>
    )
}
