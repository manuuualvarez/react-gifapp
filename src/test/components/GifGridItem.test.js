import React from 'react';
import { shallow } from 'enzyme'; 
import {  PropTypes } from 'prop-types';

// Components
import { GifGridItem } from '../../components/GifGridItem';


describe('Unit test on <GifGridItem />', () => {

    const title = 'Google';
    const url = 'http://google.com';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('Show  component succesfully', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("Shoul have a paragraph with the title", () =>{
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })

    test('Should have the image equal to src and alt of the props', () => {
        const img = wrapper.find('img');
        console.log(img.props());
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    test('Should show animations with css class', () => {
        const div = wrapper.find('div');
        expect(div.hasClass('animate__fadeOut')).toBe(true);
    })
})

GifGridItem.propTypes = { 
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}

