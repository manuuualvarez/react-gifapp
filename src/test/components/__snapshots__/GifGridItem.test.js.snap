// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test on <GifGridItem /> Show  component succesfully 1`] = `
<div
  className="card animate__animated animate__fadeOut"
>
  <img
    alt="Google"
    src="http://google.com"
  />
  <p>
     
    Google
  </p>
</div>
`;
